py/desispec/sky.py

```diff
@@ -982,25 +982,63 @@ def subtract_sky(frame, skymodel, apply_throughput_correction = False, zero_ivar
         log.error(message)
         raise ValueError(message)
 
-    if apply_throughput_correction and skymodel.throughput_corrections is not None :
-        # need to fit for a multiplicative factor of the sky model
-        # before subtraction
-        # we are going to use a set of bright sky lines,
-        # and fit a multiplicative factor + background around
-        # each of them individually, and then combine the results
-        # with outlier rejection in case a source emission line
-        # coincides with one of the sky lines.
-        for fiber in range(frame.flux.shape[0]) :
-            # apply this correction to the sky model even if we have not fit it (default can be 1 or 0)
-            skymodel.flux[fiber] *= skymodel.throughput_corrections[fiber]
+    if skymodel.throughput_corrections is not None :
+        # a multiplicative factor + background around
+        # each of the bright sky lines has been fit.
+        # here we apply this correction to the emission lines only or to the whole
+        # sky spectrum
+        if apply_throughput_correction  :
+            for fiber in range(frame.flux.shape[0]) :
+                # apply this correction to the sky model even if we have not fit it (default can be 1 or 0)
+                skymodel.flux[fiber] *= skymodel.throughput_corrections[fiber]
+
+        elif apply_throughput_correction_to_lines :
+
+            # we first identify bright lines on the sky model
+
```

Comment:
Do you really want to ~discover the line list here, versus using some line list like this one?
https://github.com/desihub/desispec/blob/main/py/desispec/sky.py#L1027-L1040
I guess likely the only difference is for nebular MW emission, though.  I don't think we ever established what the difference was between 'continuum' tpcorr and 'sky line tpcorr', so maybe one could imagine that nebular MW emission would also be better modeled here.

Reply:
Agreed. I was exactly doing this change (fixed list of lines :-) ), after finding that my current code was not robust enough. Nebular MW emission is another story.